lib/CAPI/Dialects.cpp

```diff
@@ -12,20 +12,49 @@
 #include "mlir/CAPI/IR.h"
 #include "mlir/CAPI/Registration.h"
 #include "mlir/IR/Types.h"
+#include "structured/Dialect/Indexing/IR/Indexing.h"
 #include "structured/Dialect/Iterators/IR/Iterators.h"
 #include "structured/Dialect/Tabular/IR/Tabular.h"
 #include "structured/Dialect/Tuple/IR/Tuple.h"
 #include "llvm/ADT/StringRef.h"
-
-//===----------------------------------------------------------------------===//
-// Iterators dialect and types
-//===----------------------------------------------------------------------===//
+#include <mlir-c/BuiltinTypes.h>
+#include <mlir/CAPI/Support.h>
 
 using namespace mlir;
+using namespace mlir::indexing;
 using namespace mlir::iterators;
 using namespace mlir::tabular;
 using namespace mlir::tuple;
 
+//===----------------------------------------------------------------------===//
+// Indexing dialect and attributes
+//===----------------------------------------------------------------------===//
+
+MLIR_DEFINE_CAPI_DIALECT_REGISTRATION(Indexing, indexing, IndexingDialect)
+
+bool mlirTypeIsAIndexingCustom(MlirType type) {
+  return unwrap(type).isa<CustomType>();
+}
+
+MlirType mlirIndexingCustomTypeGet(MlirContext ctx, MlirStringRef str) {
+  return wrap(CustomType::get(unwrap(ctx), unwrap(str)));
+}
+
+bool mlirIsATensorValue(MlirValue value) {
+  return mlirTypeIsATensor(mlirValueGetType(value));
+}
+
+bool mlirIsAnArithValue(MlirValue value) {
```

Comment:
Super-tiny nit: the conventions don't seem to respect English grammar and never change "a" to "an" in symbol names (e.g., `mlirTypeIsAInteger`).

Reply:
My 4th grade english teacher will be very disappointed in me...